src/drivers/intel/baytrail/ssp.c

```diff
@@ -619,16 +620,22 @@ static int ssp_probe(struct dai *dai)
 	 * - CHT shares SSP 0,1,2 IRQs with SSP 3,4,5
 	 */
 	if (ssp_irq(dai) >= IRQ_CHT_SSP_OFFSET)
-		interrupt_register(ssp_irq(dai) - IRQ_CHT_SSP_OFFSET,
-				   IRQ_AUTO_UNMASK, ssp_irq_handler, dai);
+		ret = interrupt_register(ssp_irq(dai) - IRQ_CHT_SSP_OFFSET,
+					 IRQ_AUTO_UNMASK, ssp_irq_handler, dai);
 	else
-		interrupt_register(ssp_irq(dai), IRQ_AUTO_UNMASK,
-				   ssp_irq_handler, dai);
+		ret = interrupt_register(ssp_irq(dai), IRQ_AUTO_UNMASK,
+					 ssp_irq_handler, dai);
 #else
 	/* register our IRQ handler */
-	interrupt_register(ssp_irq(dai), IRQ_AUTO_UNMASK, ssp_irq_handler,
-			   dai);
+	ret = interrupt_register(ssp_irq(dai), IRQ_AUTO_UNMASK, ssp_irq_handler,
+				 dai);
 #endif
+	if (ret < 0) {
+		trace_ssp_error("SSP failed to allocate IRQ");
+		rfree(ssp);
```

Comment:
we may want to have a test on the ssp = rzalloc upfront as well?
It beats me that such allocations are not tested for.

Reply:
@plbossart teh rzalloc in this case is using the system heap, which either returns a ptr or panics. It's meant for one time only allocs that never need to be freed.